src/visualizers/Chaos.ts

```diff
@@ -1,429 +1,804 @@
-
-        // Draw corner labels if desired
-        if (this.showLabels) {
-            this.sketch
-                .stroke(this.currentPalette.textColor)
-                .fill(this.currentPalette.textColor)
-                .strokeWeight(textStroke)
-                .textSize(textSize)
-                .textAlign(this.sketch.CENTER, this.sketch.CENTER)
-            // Get appropriate locations for the labels
-            const cornersLabels = this.chaosWindow(
-                center,
-                radius * labelOutset
+
+        // no stroke (in particular, no outline on circles)
+        this.sketch.strokeWeight(0)
+
+        // draw from beginning
+        this.cursor = 0
+
+        // set up the arrays
+        this.refresh()
+    }
+
+    // reset the computed dots (arrays)
+    // lose all the old data
+    refresh() {
+        this.chunks = markRaw([])
+        const firstSize = 1
+        const firstColor = this.sketch.color(this.bgColor)
+        // put the first walker dot into the arrays
+        this.dots = markRaw([[new p5.Vector()]])
+        this.dotsIndices = markRaw([[0n]])
+        this.dotsSizes = markRaw([[firstSize]])
+        this.dotsColors = markRaw([[firstColor]])
+        this.dotsCorners = markRaw([[0]])
+        // for every other walker, push another walker array
+        // keep in mind the first entry in the arrays is a "dummy" indicating
+        // that you start at the origin, but we don't draw that dot
+        for (let currWalker = 0; currWalker < this.walkers; currWalker++) {
+            this.dots.push([new p5.Vector()])
+            this.dotsIndices.push([0n])
+            this.dotsSizes.push([firstSize])
+            this.dotsColors.push([firstColor])
+            this.dotsCorners.push([0])
+        }
+        this.redraw()
+    }
+
+    // reset the drawing without resetting the computed dots
+    // blanks the screen and sets up to redraw the dots
+    redraw() {
+        this.cursor = 0
+        // prepare canvas with polygon labels
+        this.sketch.background(this.bgColor)
+        this.drawLabels()
+    }
+
+    drawLabels() {
+        if (!this.showLabels) return
+
+        // text appearance control
+        const shrink = Math.log(this.corners)
+        // Shrink the numbers appropriately (up to about 100 corners or so)
+        const textSize = (this.sketch.width * 0.04) / shrink
+
+        // labels are currently white: TODO make contrast background
+        this.sketch
+            .stroke('white')
+            .fill('white')
+            .strokeWeight(0)
+            .textSize(textSize)
+
+        // Find locations for labels
+        const cornersLabels = this.chaosWindow(this.radius * this.labelOutset)
+
+        // Draw the labels
+        for (let c = 0; c < this.corners; c++) {
+            const label = cornersLabels[c]
+            // WebGL is warning here, but we HAVE loaded a font back in setup??
+            this.sketch.text(String(c), label.x, label.y)
```

Comment:
the font loading is happening asynchronously, so I think it's possible that it hasn't completed by the first time that drawLabels is called,

Reply:
yes this is definitely happening, do we just leave it as is and remove the comment?

Reply:
No, I just wait to draw the labels until we have the fonts. This may not interact correctly with fade, so we may still yet have to tweak it.